File: golang/reverse_integer.go
package main

import (
	"fmt"
	"strconv"
)

func reverse(x int) int {
	var value int
	var ret int
	if x >= 0 {
		value = x
	} else {
		value = -x
	}

	str := strconv.Itoa(value)
	var arr []byte
	for i := len(str) - 1; i >= 0; i-- {
		arr = append(arr, str[i])
	}
	ret, _ = strconv.Atoi(string(arr))

	if x >= 0 {
		ret = ret
	} else {
		ret = -ret
	}

	if ret >= 2147483647 || ret <= -2147483648 {
		ret = 0
	}

	return ret
}

func main() {
	res := reverse(-2147483648)
	fmt.Println(res)
}

File: c/reverse_integer.c
#include <stdio.h>
#include <math.h>

int reverse(int x)
{
	int operateNo = 0;
	if (x >= 0) {
		operateNo = x;
	} else {
		operateNo = -x;
	}

	int quotient = operateNo / 10;
	int reminder = operateNo % 10;
	int arr[10] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};

	int count = 0;
	while (quotient > 0) {
		arr[count] = reminder;
		reminder = quotient % 10;
		quotient = quotient / 10;
		count++;
	}

	arr[count] = reminder;

	int figure = 0;
	int i;

	for (i = 0; i < 10; ++i) {
		printf("arr: %d\n", arr[i]);
	}

	int first = 0;
	int last = 0;
	for (i = 0; i < 10; ++i) {
		if (arr[i] != 0) {
			first = i;
			break;
		}
	}

	for (i = 0; i < 10; ++i) {
		if (arr[i] != 0) {
			last = i;
		}
	}

	figure = last - first;

	int res = 0;

	for (i = first; i <= last; i++) {
		res += arr[i] * pow(10, figure);
		figure -= 1;
	}

	if (res < 0) {
		return 0;
	}

	int resfinal = 0;
	if (x >= 0) {
		resfinal = res;
	} else {
		resfinal = -res;
	}

	if (resfinal > pow(2, 32) - 1 || resfinal < -(pow(2, 32))) {
		return 0;
	} else {
		return resfinal;
	}
}

int main() 
{
	int test = 1534236469;
	int res = reverse(test);

	printf("%d\n", res);
	return 0;
}